### PR TITLE
[v2.7] Refactor hosted provisioning tests

### PR DESCRIPTION
**Pull request by markusewalker**
## Problem

Recently, we refactored the `provisioning` folder at a whole. However, we did not apply the same changes to the `hosted` sub-folder. Although team 2 does not directly work with hosted clusters anymore, we should still ensure consistency across the board for the tests we did maintain.
 
## Solution

Refactored the `hosted` folder to match the broader `provisioning` folder changes seen in RKE1, RKE2 and K3S.
 
### Manual Testing

Locally ran AKS and EKS with the changes; validated that the tests ran successfully as expected.

### Automated Testing

Jenkins runs will be provided to reviewers offline.

### CODE REVIEW

**Comment by igomez06** [comment]
@markusewalker can you create a forward port please?

File: tests/v2/validation/nodescaling/scaling_node_driver_aks_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+}
+
+func (s *AKSNodeScalingTestSuite) TestScalingAKSNodePools() {
+	var oneNode int64 = 1
+	var twoNodes int64 = 2
```
Comment (caliskanugur) [inline tests/v2/validation/nodescaling/scaling_node_driver_aks_test.go:42]: This seems missing in 2.8, is it expected?

File: tests/v2/validation/nodescaling/scaling_node_driver_eks_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+}
+
+func (s *EKSNodeScalingTestSuite) TestScalingEKSNodePools() {
+	var oneNode int64 = 1
+	var twoNodes int64 = 2
```
Comment (caliskanugur) [inline tests/v2/validation/nodescaling/scaling_node_driver_eks_test.go:42]: Same for this

File: tests/v2/validation/nodescaling/scaling_node_driver_gke_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+}
+
+func (s *GKENodeScalingTestSuite) TestScalingGKENodePools() {
+	var oneNode int64 = 1
+	var twoNodes int64 = 2
```
Comment (caliskanugur) [inline tests/v2/validation/nodescaling/scaling_node_driver_gke_test.go:42]: And same for this too

File: tests/v2/validation/nodescaling/scaling_node_driver_rke1_test.go [hunk 4]
```diff
@@ -80,10 +80,14 @@ func (s *RKE1NodeScalingTestSuite) TestScalingRKE1NodePools() {
 }
 
 func (s *RKE1NodeScalingTestSuite) TestScalingRKE1NodePoolsDynamicInput() {
+	if s.scalingConfig.NodePools.NodeRoles == nil {
+		s.T().Skip()
+	}
```
Comment (caliskanugur) [inline tests/v2/validation/nodescaling/scaling_node_driver_rke1_test.go:85]: Somehow there are extra lines for this on the 2.8 PR

File: tests/v2/validation/nodescaling/scaling_node_driver_aks_test.go (hunk 1 above)
Comment (igomez06) [reply tests/v2/validation/nodescaling/scaling_node_driver_aks_test.go:42]: v2.8 has them in a single file

Comment (caliskanugur) [reply tests/v2/validation/nodescaling/scaling_node_driver_aks_test.go:42]: Oh makes sense, so 2.7 is not updated right? @markusewalker. I missed those additions on 2.8 PR

Comment (markusewalker) [reply tests/v2/validation/nodescaling/scaling_node_driver_aks_test.go:42]: yeah I need to update this branch, I'll do that now.

**Review by caliskanugur — APPROVED** [review]
Looks good to me

**Review by igomez06 — APPROVED** [review]
LGTM

**Pull request merged**